### DISCUSSION TITLE
docs/core/gdb-context

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#41] [General]
# docs/core/gdb-context

本文介绍了在GoFrame框架中如何通过ORM支持自定义的上下文变量，以实现异步IO控制、链路跟踪、嵌套事务等功能。通过使用Ctx方法，开发者可以轻松传递自定义上下文变量，实现更复杂的请求控制和跟踪。文章提供了请求超时控制和模型上下文操作的具体示例和建议。

https://goframe.org/docs/core/gdb-context

**Comment by lankii** [comment]
type DISTINCTKey struct {
		gmeta.Meta `orm:"table:telemetry_datas"`
		Id         string
		Dkey       string
	}
	dskey := []DISTINCTKey{}
	var keylist []string
	var idlist []string

结构体如上，特意指定了表，照样产生查询id表的错误，啥情况
err := mk.Scan(&dskey)
		if err != nil {
			fmt.Println("转换错误", err)
		}
执行Scan就报错

2025-07-01T14:03:47.669+08:00 [ERRO] {b46095a3630b4e189b110d5c69fbb86c} [  1 ms] [default] [gfast] [rows:0  ] SHOW FULL COLUMNS FROM `id`  这是表中的字段
Error: Error 1146 (42S02): Table 'gfast.id' doesn't exist
Stack:
1.  github.com/lankiiIOT/internal/controller/v1.(*cProductModel).TelemetryDatasListForPage
    E:/goFIOT/gfast-os-v3.2/internal/controller/v1/v1_dataTelemetry.go:61
2.  github.com/lankiiIOT/internal/cmd.MiddlewareCORS
    E:/goFIOT/gfast-os-v3.2/internal/cmd/cmd.go:92
